core/jvm/src/main/scala/zio/zmx/internal/ConcurrentSummary.scala

```diff
@@ -31,9 +29,10 @@ object ConcurrentSummary {
 
   def manual(maxSize: Int, maxAge: Duration, error: Double, quantiles: Chunk[Double]): ConcurrentSummary =
     new ConcurrentSummary {
-      private val values          = new ConcurrentLinkedDeque[(java.time.Instant, Double)]
+      private val values = new Array[(java.time.Instant, Double)](maxSize)
```

Comment:
Think you want `AtomicReferenceArray` since each element needs to be updated atomically.